content/articles/an-extensive-guide-on-working-with-images-in-django/index.md

```diff
@@ -0,0 +1,424 @@
+```bash
+cd Images
+```
+```bash
+py -m venv .venv
+```
+```bash
+.venv\Scripts\activate.bat
+```
+Now we will need to install:
+ - Django - this python package helps build our website using the Django web development framework.
+ - Pillow - this is a python image library that we shall use to manipulate our images.
+
+We will install the above using:
+```bash
+pip install django
+```
+```bash
+pip install pillow
+```
+After the installation, we can now proceed to create the Django project named `myGallery`:
+```cd 
+django-admin startproject myGallery
+```
+Enter the project directory:
+```bash
+cd myGallery
+```
+and then create our app `demo`:
+```bash
+py manage.py startapp demo
+```
+Now we need to register our `demo` app in the list of installed apps. In the `settings.py` file under `installed apps`:
+
+```py
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'demo',                        # new
+]
+```
+Now we need to tell Django which URL to use when serving the media files and also give it a path to the root directory where our images will be stored. We shall do this by using `MEDIA_URL` and `MEDIA_ROOT` respectively. So while still in `settings.py` we shall add the following:
+
+```py
+MEDIA_URL = '/media/'
+MEDIA_ROOT = os.path.join(BASE_DIR, 'media')
+
+```
+Also, make sure to import os at the top of `settings.py`
+```py
+import os
+```
+Now to enable a user to upload images during development, we need to add the following settings in our project-level `urls.py` for testing purposes. It should look like this:
+
+```py 
+from django.contrib import admin
+from django.urls import path
+from django.conf import settings
+from django.conf.urls.static import static 
+
+urlpatterns = [
+    path('admin/', admin.site.urls),
+
+] + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+
+```
+We are now done laying down the groundwork of uploading our images.
+
+### Step 2 - Creating and registering the models
+We can now proceed and define a model that shall be used for storing the images in our app. In `settings.py` we shall create a model with two fields, title of a photo and the photo itself:
+```py
+from django.db import models
+
+class Image(models.Model):
+    title = models.CharField(max_length=20)
+    photo = models.ImageField(upload_to='pics')
+```
+
+The `upload_to` tells Django to store the photo in a directory called `pics` which shall be created in the `media` directory upon image upload.
+
+Now we can register our model in `admin.py`:
+```py
+from django.contrib import admin
+from .models import Image
+
+class imageAdmin(admin.ModelAdmin):
+    list_display = ["title", "photo"]
+
+admin.site.register(Image, imageAdmin)
+```
+The `list_display` list tells Django admin what to display in the admin dashboard.
```

Comment:
So, all the images and it's respective titles are displayed in the dashboard? Mention that